sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/DefaultSource.scala

```diff
@@ -63,7 +63,9 @@ class DefaultSource extends FileFormat with DataSourceRegister {
     val header = if (csvOptions.headerFlag) {
       firstRow
     } else {
-      firstRow.zipWithIndex.map { case (value, index) => s"C$index" }
+      // firstRow.zipWithIndex.map { case (value, index) => s"C$index" }
+      new BulkCsvReader(rdd.toLocalIterator, csvOptions, null).next()
+        .zipWithIndex.map { case (value, index) => s"C$index" }
```

Comment:
Also, I think this is related with [`UnescspedQuoteHandling`](https://github.com/apache/spark/blob/ac12b35d31ef1d1663511bf6ae826a9cc0278f20/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVParser.scala#L50).